src/Microsoft.Diagnostics.Monitoring/DiagnosticsEventPipeProcessor.cs

```diff
@@ -33,24 +34,27 @@ public class DiagnosticsEventPipeProcessor : IAsyncDisposable
         private readonly PipeMode _mode;
         private readonly int _metricIntervalSeconds;
         private readonly LogLevel _logsLevel;
+        private readonly Action<string> _processInfoCallback;
 
         public DiagnosticsEventPipeProcessor(
             PipeMode mode,
             ILoggerFactory loggerFactory = null,
             IEnumerable<IMetricsLogger> metricLoggers = null,
             int metricIntervalSeconds = 10,
             MemoryGraph gcGraph = null,
-            LogLevel logsLevel = LogLevel.Debug)
+            LogLevel logsLevel = LogLevel.Debug,
+            Action<string> processInfoCallback = null)
```

Comment:
Could you put some comments here to organize which inputs are correlated with which outputs? I understand we'll be refactoring this later, but I think this guy is getting out of hand :-)